src/Control/Functor/Linear/Internal/Class.hs

```diff
@@ -147,3 +162,28 @@ foldM :: forall m a b. Monad m => (b %1-> a %1-> m b) -> b %1-> [a] %1-> m b
 foldM _ i [] = return i
 foldM f i (x:xs) = f i x >>= \i' -> foldM f i' xs
 
+
+data Zero a = Zero
+data One a = One a
+class GFunctor i o f where
+  gmap :: i (a %1-> b) %1-> f a %1-> (f b, o (a %1-> b))
```

Comment:
I'm not super convinced by the state-monad-like threading of `f`, to be honest. It would be preferable to figure out, in a product, where the `f` should go based on some type class magic.

Reply:
Back to the drawing board!